src/UpsilonGC/UpsilonGCHeap.cpp

```diff
@@ -618,3 +630,24 @@ void UpsilonGCHeap::registerSegment(uint8_t* new_pages)
 {
 	segments[segmentsCount++] = new_pages;
 }
+
+#ifdef __linux__
+uint64_t UpsilonGCHeap::GetTotalAllocatedBytes()
+{
+    return 0;
+}
+
+int UpsilonGCHeap::GetLastGCPercentTimeInGC()
+{
+    return 0;
+}
+
+size_t UpsilonGCHeap::GetLastGCGenerationSize(int gen)
+{
+    return 0;
+}
+
+void UpsilonGCHeap::DiagWalkObject2(Object* obj, walk_fn2 fn, void* context)
+{
+}
+#endif
```

Comment:
Is this because you need to support newer runtime in your case? If so, I would like to avoid `#ifdef __linux__` and just bump up the supported GC API version by the Upsilon GC (no matter the operating system).